satpy/readers/viirs_edr.py

```diff
@@ -343,3 +358,19 @@ def _scale_data(self):
             add_offset = self.nc[self._manual_scalings[var_name][1]]
             data_arr.data = data_arr.data * scale_factor.data + add_offset.data
             self.nc[var_name] = data_arr
+
+
+class VIIRSAODHandler(VIIRSJRRFileHandler):
+    """File handler for AOD data files."""
+
+    def __init__(self, *args, aod_qc_filter: int | None = None, **kwargs) -> None:
+        """Initialize file handler and keep track of QC filtering."""
+        super().__init__(*args, **kwargs)
+        self._aod_qc_filter = aod_qc_filter
+
+    def _mask_invalid(self, data_arr: xr.DataArray, ds_info: dict) -> xr.DataArray:
+        new_data_arr = super()._mask_invalid(data_arr, ds_info)
+        if self._aod_qc_filter is None or ds_info["name"] != "AOD550":
+            return new_data_arr
+        qc_all = self.nc["QCAll"]
+        return new_data_arr.where(qc_all <= self._aod_qc_filter)
```

Comment:
Do I understand correctly from the doctring hight up that the qc_all will have values set to 0 for high quality pixels, 1 for medium quality pixels, and so on? In that case, shouldn't it be `new_data_arr.where(qc_all >= self._aod_qc_filter)`?

Sorry if this is a stupid question, my brain is probably not up to speed yet.

Reply:
Xarray's `.where` is backwards. The condition is where we want valid data.